packages/kbn-server-http-tools/src/set_tls_config.ts

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import type { Server as HapiServer } from '@hapi/hapi';
+import type { Server as HttpServer } from 'http';
+import type { Server as TlsServer } from 'https';
+import type { ISslConfig } from './types';
+import { getServerTLSOptions } from './get_server_options';
+
+function isServerTLS(server: HttpServer): server is TlsServer {
+  return 'setSecureContext' in server;
+}
```

Comment:
NIT: I could probably have used `server instanceof TlsServer` instead, but tbh this works and is easier to stub for testing.